Source/MediaStorageAndFileFormat/gdcmImageHelper.cxx

```diff
@@ -1451,8 +1463,24 @@ std::vector<double> ImageHelper::GetSpacingValue(File const & f)
       return sp;
       }
     }
+  else if( ms == MediaStorage::SecondaryCaptureImageStorage )
+    {
+      if( ds.FindDataElement( Tag(0x0028,0x0030) ) )
+        {
+        // Type 1C in 'SC Image' (for calibrated images)
+        spacingtag = Tag(0x0028,0x0030);
+        }
+      else if( ds.FindDataElement( Tag(0x0018,0x2010) ) )
+        {
+        // Type 3 in 'SC Image'
+        spacingtag = Tag(0x0018,0x2010);
+        }
+    }
+  else
+    {
+    spacingtag = GetSpacingTagFromMediaStorage(ms);
+    }
 
-  Tag spacingtag = GetSpacingTagFromMediaStorage(ms);
```

Comment:
@malaterre that makes sense.

How do you prefer to perform the state checks,

```
ds.FindDataElement( Tag(0x0028,0x0030) )
```

```
ds.FindDataElement( Tag(0x0018,0x2010) )
```

in `GetSpacingTagFromMediaStorage`? For example, two new boolean optional arguments? `have00280030`, `have00180030`? Pass `ds` as a second argument?

Reply:
My understanding of the logic is:

1. 0x0028,0x0030 if present, else
2. 0x0018,0x2010 if present, else
3. 0xffff,0xffff if present

and we would need to update `GetSpacingTagFromMediaStorage` to check the "if present" for those tags if we want to move the logic there. Otherwise, `GetSpacingTagFromMediaStorage` is unchanged.

@issakomi is that correct?

Reply:
@thewtex  Thank you!

The logic is here: [Basic Pixel Spacing Calibration Macro Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_10.7.html#table_10-10) is part of the [SC Image Module Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.8.6.2.html#table_C.8-2)

I intentionally didn't update `GetSpacingTagFromMediaStorage` at this point for simplicity.

> P.S. The PR intentionally doesn't change writing of SC files.

Reply:
@issakomi excellent, thank you for the links and clarification!

@malaterre I added a patch to the tip that uses `GetSpacingTagFromMediaStorage` by default (previous behavior) and only uses (0028,0030) when present when `SecondaryCaptureImagePlaneModule` is `true`.

Reply:
Pixel Spacing (0028,0030) is also in the [Image Plane Module](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.2.html#sect_C.7.6.2). Probably in this particular dataset it is part of the Image Plane Module (extension). If you think there should be a more accurate check, then why not, of course.